src/core_plugins/elasticsearch/index.js

```diff
@@ -47,7 +47,10 @@ export default function (kibana) {
 
       return object({
         enabled: boolean().default(true),
-        url: string().uri({ scheme: ['http', 'https'] }).default('http://localhost:9200'),
```

Comment:
Unless we want to wait until 7.0 for this, I don't think we can instantly deprecate this setting. I wonder if we can maintain backwards compatibility with Joi.

Something like:

```
url: string().uri({ scheme: ['http', 'https'] }).default('http://localhost:9200'),
hosts: array().items(string().uri({ scheme: ['http', 'https'] })).single().default(ref('url')),
```

Reply:
Jon pointed out this should be handled by the deprecation logic, however, there is currently a bug with it which will be resolved by https://github.com/elastic/kibana/pull/21294

That should be fine, but we want to make sure that we maintain compatibility for programatic access to `elasticsearch.url`. We can return just the first url in the hosts.